docs/modules/ROOT/pages/quarkus-azure-cosmos.adoc

```diff
@@ -0,0 +1,218 @@
+    --account-name kvquarkusazurecosmos080824 \
+    --resource-group rg-quarkus-azure-cosmos \
+    --scope "/" \
+    --principal-id @- \
+    --role-definition-id 00000000-0000-0000-0000-000000000002
+----
+
+Notice that you cannot use any Azure Cosmos DB data plane SDK to authenticate management operations with a Microsoft Entra identity, you need to create database and container manually.
+Run the following commands to create a database `demodb` and a container `democontainer` using Azure CLI.
+
+[source,shell]
+----
+az cosmosdb sql database create \
+    -a kvquarkusazurecosmos080824 \
+    -g rg-quarkus-azure-cosmos \
+    -n demodb
+az cosmosdb sql container create \
+    -a kvquarkusazurecosmos080824 \
+    -g rg-quarkus-azure-cosmos \
+    -d demodb \
+    -n democontainer \
+    -p "/id"
+----
+
+=== Configure the Azure Cosmos DB Client
+
+As you can see below in the _Extension Configuration Reference_ section, the property `quarkus.azure.cosmos.endpoint` is required if the extension is enabled.
+To get the endpoint of the Azure Cosmos DB account, execute the following Azure CLI command:
+
+[source,shell]
+----
+export QUARKUS_AZURE_COSMOS_ENDPOINT=$(az cosmosdb show \
+    -n kvquarkusazurecosmos080824 \
+    -g rg-quarkus-azure-cosmos \
+    --query documentEndpoint -o tsv)
+echo "QUARKUS_AZURE_COSMOS_ENDPOINT is: ${QUARKUS_AZURE_COSMOS_ENDPOINT}"
+----
+
+Notice that you get the connection string and set it to environment variable `QUARKUS_AZURE_COSMOS_ENDPOINT`, instead of setting it to property `quarkus.azure.cosmos.endpoint` in the `application.properties` file.
+
+Although technically both approaches work, using environment variable is recommended and more secure as there's no risk of committing the connection string to source control.
+
+=== Inject the Azure Cosmos DB Client
+
+Now that your Azure environment is ready and you have configured the extension, you can inject the `com.azure.cosmos.CosmosClient` object in your imperative application or inject the `com.azure.cosmos.CosmosAsyncClient` object in your reactive application, so you can interact with Azure Cosmos DB.
+
+==== Use the CosmosClient in an imperative application
+
+The `createItem` method first creates the item with request payload in the specified database and container of the Azure Cosmos DB account.
+The `getItem` method reads the item with the id from the specified database and container of the Azure Cosmos DB account.
+
+[source,java]
+----
+@Path("/quarkus-azure-cosmos")
+@ApplicationScoped
+public class CosmosResource {
+
+    @Inject
+    CosmosClient cosmosClient;
+
+    @Path("/{database}/{container}")
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response createItem(
+            @PathParam("database") String database,
+            @PathParam("container") String container,
+            Item body,
+            @Context UriInfo uriInfo) {
+
+        getContainer(database, container).upsertItem(body);
```

Comment:
Where is `getContainer` defined?

Reply:
Good catch. Fixed in a249fe4d89b84137f7379b6eea30708b8a9a4d1b.